File: node_api/src/model/github/GithubRelease.ts
export default class GithubRelease {
    version: string;
    title: string;
    createdDate: Date;
    publishedDate: Date;
    releaseNotes: string[];
    author: string;
    url: string

    constructor(version: string, title: string, createdDate: Date, publishedDate: Date, releaseNotes: string, author: string, url: string) {
        this.version = version;
        this.title = title;
        this.createdDate = createdDate;
        this.publishedDate = publishedDate;
        this.releaseNotes = this.processReleaseNotes(releaseNotes);
        this.author = author;
        this.url = url;
    }

    processReleaseNotes(releaseNotes: string): string[] {
      return releaseNotes.split('\r\n').map(item => item.replaceAll("-   ", "").trim());
    }
}

File: src/main/java/org/endeavourhealth/imapi/model/github/GithubRelease.java
package org.endeavourhealth.imapi.model.github;

import lombok.Getter;

import java.util.List;

@Getter
public class GithubRelease {
  String version;
  String title;
  String createdDate;
  String publishedDate;
  List<String> releaseNotes;
  String author;
  String url;

  public GithubRelease(String version, String title, String createdDate, String publishedDate, List<String> releaseNotes, String author, String url) {
    this.version = version;
    this.title = title;
    this.createdDate = createdDate;
    this.publishedDate = publishedDate;
    this.releaseNotes = releaseNotes;
    this.author = author;
    this.url = url;
  }

  public GithubRelease() {}

  public GithubRelease setVersion(String version) {
    this.version = version;
    return this;
  }

  public GithubRelease setTitle(String title) {
    this.title = title;
    return this;
  }

  public GithubRelease setCreatedDate(String createdDate) {
    this.createdDate = createdDate;
    return this;
  }

  public GithubRelease setPublishedDate(String publishedDate) {
    this.publishedDate = publishedDate;
    return this;
  }

  public GithubRelease setReleaseNotes(List<String> releaseNotes) {
    this.releaseNotes = releaseNotes;
    return this;
  }

  public GithubRelease setAuthor(String author) {
    this.author = author;
    return this;
  }

  public GithubRelease setUrl(String url) {
    this.url = url;
    return this;
  }
}
